### ISSUE TITLE
[Starbook V/VI - AMD] Can't Install Coreboot / No Supported Device

### ISSUE THREAD

**Issue by 3lpsy** [StarLabsLtd/firmware#65]
## Background

I ordered a Starbook MK V last may so I'm not sure if I own a V or VI (it has a fingerprint reader if that helps). While I ordered it with Coreboot, it came with American Megatrends. 

### Other Info
CPU: AMD Ryzen 7 5800U with Radeon Graphics
OS: Arch Linux

## First Attempt
When attempting to switch branches, I first performed a BIOS update and rebooted. Then when running `switch-branch`, i received a No Supported Devices message. I am running Arch Linux with the standard `linux` kernel (I had different issues under `linux-hardened` but I believe those are expected). I first attempted to use the Manjaro `fwupd-starlabs` and f`lashrom-starlabs` packages. I've included the details below

### Arch Linux + Standard Kernel + Starlabs fwupdmgr + Starlabs Flashrom
```
$ sudo fwupdmgr switch-branch
No supported devices

$ sudo flashrom -p internal
flashrom unknown on Linux 6.1.8-arch1-1 (x86_64)
flashrom is free software, get the source code at https://flashrom.org

Using clock_gettime for delay loops (clk_id: 1, resolution: 1ns).
No DMI table found.
Found chipset "AMD FP4".
Enabling flash write... FCH device found but SMBus revision 0x51 does not match known values.
Please report this to flashrom@flashrom.org and include this log and
the output of lspci -nnvx, thanks!.
Could not determine chipset generation.PROBLEMS, continuing anyway
========================================================================
You may be running flashrom on an unknown laptop. We could not
detect this for sure because your vendor has not set up the SMBIOS
tables correctly. Some internal buses have been disabled for
safety reasons. You can enforce using all buses by adding
  -p internal:laptop=this_is_not_a_laptop
to the command line, but please read the following warning if you
are not sure.

Laptops, notebooks and netbooks are difficult to support and we
recommend to use the vendor flashing utility. The embedded controller
(EC) in these machines often interacts badly with flashing.
See the manpage and https://flashrom.org/Laptops for details.

If flash is shared with the EC, erase is guaranteed to brick your laptop
and write may brick your laptop.
Read and probe may irritate your EC and cause fan failure, backlight
failure and sudden poweroff.
You have been warned.
========================================================================
No EEPROM/flash device found.
Note: flashrom can never write if the flash chip isn't found automatically.
```

## Second Attempt
I also installed the `flashrom-git` package from the Arch AUR and attempted to switch to coreboot.

### Arch Linux + Standard Kernel + Starlabs fwupdmgr + flashrom-git
```
$ sudo fwupdmgr switch-branch
No supported devices

$ sudo flashrom -p internal
flashrom unknown on Linux 6.1.8-arch1-1 (x86_64)
flashrom is free software, get the source code at https://flashrom.org

Using clock_gettime for delay loops (clk_id: 1, resolution: 1ns).
No DMI table found.
Found chipset "AMD FP4".
Enabling flash write... FCH device found but SMBus revision 0x51 does not match known values.
Please report this to flashrom@flashrom.org and include this log and
the output of lspci -nnvx, thanks!.
Could not determine chipset generation.PROBLEMS, continuing anyway
========================================================================
You may be running flashrom on an unknown laptop. We could not
detect this for sure because your vendor has not set up the SMBIOS
tables correctly. Some internal buses have been disabled for
safety reasons. You can enforce using all buses by adding
  -p internal:laptop=this_is_not_a_laptop
to the command line, but please read the following warning if you
are not sure.

Laptops, notebooks and netbooks are difficult to support and we
recommend to use the vendor flashing utility. The embedded controller
(EC) in these machines often interacts badly with flashing.
See the manpage and https://flashrom.org/Laptops for details.

If flash is shared with the EC, erase is guaranteed to brick your laptop
and write may brick your laptop.
Read and probe may irritate your EC and cause fan failure, backlight
failure and sudden poweroff.
You have been warned.
========================================================================
No EEPROM/flash device found.
Note: flashrom can never write if the flash chip isn't found automatically.
```

## Third Attempt
Finally, I installed Manjaro on a USB device, booted into it and tried performing the same action. I of course used the starlabs fwupd and flashrom packages. I don't have the raw results for the flashrom command, but the switch-branch command had the same output. 

Is Coreboot supported on the AMD variant? And if possible, can you please advise me on what I'm missing

**Comment by rmsthebest** [comment]
They are waiting for an FSP fix from AMD before they upload coreboot for the AMD version. Support has said everyone who bought the coreboot amd laptops will be notified when it is available.

Imho it is strange that each customer has to discover for themselves that they didnt receive what they purchased, but it is what it is I guess. To be clear, I'm glad they sent them even if they havent got coreboot working yet, but the delivery receipt should not have said that it contained a laptop with coreboot.

**Comment by 3lpsy** [comment]
Yeah, I'm just happy to have my laptop. Fine being patient and glad support will eventually get added :)

**Comment by neeshy** [comment]
Any news on this? It's been over 5 months. Have you guys received word from AMD about that FSP fix? On our end, is it currently possible to build coreboot from source and flash it manually w/ flashrom?

**Comment by Sean-StarLabs** [comment]
Yes, still waiting for a new build though. The latest emails hint at getting something soon.

> On our end, is it currently possible to build coreboot from source and flash it manually w/ flashrom?

You can build and flash anything you like; it wouldn't circumvent the problem and would introduce others that exist in the publicly available blobs.

> As far as I can tell, only Intel support for the StarBooks has been merged into upstream coreboot. Likewise, I only see Intel related commits on StarLab's coreboot fork.

Patches won't be merged until the problem is fixed.

**Comment by neeshy** [comment]
> The latest emails hint at getting something soon.

That sounds promising :)

**Comment by sysnux** [comment]
> Yes, still waiting for a new build though. The latest emails hint at getting something soon.

Any update?

**Comment by Sean-StarLabs** [comment]
It's basically being held up by internal AMD stuff (NDA's prevent too much detail there) - I'll give them another nudge!

**Comment by sysnux** [comment]
Still waiting for the coreboot I ordered more than one year ago...  fortunately, the StarBook AMD arrived sooner ;)

**Comment by flokaiser** [comment]
Do you have any update?

**Comment by flokaiser** [comment]
I think we were ghosted by a company :(

**Comment by Sean-StarLabs** [comment]
Sorry, not sure how I missed that - nothing yet. I'll give them another nudge.

**Comment by cyb3rmonk** [comment]
@Sean-StarLabs did you get an answer?

**Comment by Sean-StarLabs** [comment]
Yeah, so, paraphrasing - official channels are proving useless, so the engineer is going to try to reach someone from the PSP team directly. And, to be fair, that's how things get done with Intel - the only difference is "we" know the Intel devs.

> Also, I hope you guys review this mess internally. 

We're not doing any more AMDs until they fix it (for consumers).

>  I understand you depend on AMD, but if that's the case then why advertise coreboot compatibility for a setup before you get everything you need from them?

Because the code is 99% there. The way we usually do it, make the AMI firmware as it's way, way more time consuming (code base is about 20x the size of coreboot), and then do coreboot with essentially the same code. The same happened here, and it was a small details than turned into a big blocker. And, getting zero support from AMD honestly wasn't anticipated. 

> At some point you have to give up on the problem and embrace the fact that you have dozens of new friends

I like the idea, but no, I don't like giving up. AMD are slowly making progress with their open source code, and there is one other thread I'm pulling on at the moment - long shot, but worth a try.

Either way, it'll happen - the variable is when.

**Comment by hicaru** [comment]
> Yeah, so, paraphrasing - official channels are proving useless, so the engineer is going to try to reach someone from the PSP team directly. And, to be fair, that's how things get done with Intel - the only difference is "we" know the Intel devs.
> 
> > Also, I hope you guys review this mess internally.
> 
> We're not doing any more AMDs until they fix it (for consumers).
> 
> > I understand you depend on AMD, but if that's the case then why advertise coreboot compatibility for a setup before you get everything you need from them?
> 
> Because the code is 99% there. The way we usually do it, make the AMI firmware as it's way, way more time consuming (code base is about 20x the size of coreboot), and then do coreboot with essentially the same code. The same happened here, and it was a small details than turned into a big blocker. And, getting zero support from AMD honestly wasn't anticipated.
> 
> > At some point you have to give up on the problem and embrace the fact that you have dozens of new friends
> 
> I like the idea, but no, I don't like giving up. AMD are slowly making progress with their open source code, and there is one other thread I'm pulling on at the moment - long shot, but worth a try.
> 
> Either way, it'll happen - the variable is when.

<img width="792" alt="image" src="https://github.com/StarLabsLtd/firmware/assets/21293723/6dec294c-74c7-43e7-a0ad-5c4091c6bcbf">

You don't need to lie! You still have them on sale!

**Comment by Sean-StarLabs** [comment]
> You don't need to lie! You still have them on sale!

No, sorry, no idea how you got that from what I said. To clarify, we're not going to stop selling the ones we've already made because 100K of code doesn't have the license we want!

**Comment by hicaru** [comment]
> > You don't need to lie! You still have them on sale!
> 
> No, sorry, no idea how you got that from what I said. To clarify, we're not going to stop selling the ones we've already made because 100K of code doesn't have the license we want!

Could you publish it? We will build and flash it directly to chip.

**Comment by Sean-StarLabs** [comment]
Publish the coreboot patch? It's in Gerrit, but the problem isn't with building or flashing it - it's with the PSP not allowing the Ap base to be set, which stops FSP from running.

**Comment by hicaru** [comment]
> Publish the coreboot patch? It's in Gerrit, but the problem isn't with building or flashing it - it's with the PSP not allowing the Ap base to be set, which stops FSP from running.

have we any chance to have `coreboot` then in near time?

**Comment by sjlongland** [comment]
> > Publish the coreboot patch? It's in Gerrit, but the problem isn't with building or flashing it - it's with the PSP not allowing the Ap base to be set, which stops FSP from running.
> 
> have we any chance to have `coreboot` then in near time?

It's frustrating, but this is one of these situations where we're all going to have to be patient.  Good things come to those who wait and all that.

AMD's a big company, likely there's a lot of layers of bureaucracy that need to be worked around… and this takes time.  Starlabs is a UK-based company if I'm not mistaken, and AMD US-based, so time-zones further extend how long these negotiations will take.

Other equipment has been made in the past running CoreBoot on AMD hardware, so I don't think the issue is necessarily impossible to fix technically.

It'd be nice if the website had flagged that AMD + Coreboot was still being worked on (with no hard ETA known) and that the product would ship with AMI UEFI instead, but I am certain it'll happen eventually.  I'm fine waiting as this machine has been a much better out-of-box experience running Linux than most machines with similar-age hardware.  The machine still boots running UEFI firmware, and once `grub` loads the kernel and hands over, UEFI quickly becomes irrelevant anyway.

Let's give them the time they need before stomping our feet. :-)

**Comment by hicaru** [comment]
> > > Publish the coreboot patch? It's in Gerrit, but the problem isn't with building or flashing it - it's with the PSP not allowing the Ap base to be set, which stops FSP from running.
> > 
> > 
> > have we any chance to have `coreboot` then in near time?
> 
> It's frustrating, but this is one of these situations where we're all going to have to be patient. Good things come to those who wait and all that.
> 
> AMD's a big company, likely there's a lot of layers of bureaucracy that need to be worked around… and this takes time. Starlabs is a UK-based company if I'm not mistaken, and AMD US-based, so time-zones further extend how long these negotiations will take.
> 
> Other equipment has been made in the past running CoreBoot on AMD hardware, so I don't think the issue is necessarily impossible to fix technically.
> 
> It'd be nice if the website had flagged that AMD + Coreboot was still being worked on (with no hard ETA known) and that the product would ship with AMI UEFI instead, but I am certain it'll happen eventually. I'm fine waiting as this machine has been a much better out-of-box experience running Linux than most machines with similar-age hardware. The machine still boots running UEFI firmware, and once `grub` loads the kernel and hands over, UEFI quickly becomes irrelevant anyway.
> 
> Let's give them the time they need before stomping our feet. :-)

have to sit with a AMD backdoor and wait for until someone reverse it or official AMD answer.

**Comment by Sean-StarLabs** [comment]
> AMD backdoor

What do you mean?

**Comment by hicaru** [comment]
> > AMD backdoor
> 
> What do you mean?

i mean: AMD PSP  is a backdoor like Intel ME

**Comment by Sean-StarLabs** [comment]
Okay. coreboot doesn't do anything different with the PSP than AMI does.

**Comment by sjlongland** [comment]
> > > AMD backdoor
> > 
> > What do you mean?
> 
> i mean: AMD PSP is a backdoor like Intel ME

Unless you've got the equipment to non-destructively inspect every integrated circuit on your computer, I'd wager that any computer you buy or make yourself could theoretically have a backdoor you don't know about or control.

The only way around that would be to use such low-level building blocks (TTL chips, CPLDs, maybe FPGAs), that making a practical backdoor becomes impossible.  Modern off-the-shelf x86 is not for you.

If that's not appealing, then we have to accept that we're not going to "see" everything that goes on inside every IC, and some of them have "features" we may not like.  It's a package deal.  That's life.

**Comment by hicaru** [comment]
> > > > AMD backdoor
> > > 
> > > 
> > > What do you mean?
> > 
> > 
> > i mean: AMD PSP is a backdoor like Intel ME
> 
> Unless you've got the equipment to non-destructively inspect every integrated circuit on your computer, I'd wager that any computer you buy or make yourself could theoretically have a backdoor you don't know about or control.
> 
> The only way around that would be to use such low-level building blocks (TTL chips, CPLDs, maybe FPGAs), that making a practical backdoor becomes impossible. Modern off-the-shelf x86 is not for you.
> 
> If that's not appealing, then we have to accept that we're not going to "see" everything that goes on inside every IC, and some of them have "features" we may not like. It's a package deal. That's life.

To my deepest regret, I agree. There are indeed some open-source chips like RISC-V that are sufficiently open to be utilized. However, they are too weak for everyday use. Additionally, there is Elbrus, but it is overly proprietary and likely contains backdoors.

**Comment by sjlongland** [comment]
> To my deepest regret, I agree. There are indeed some open-source chips like RISC-V that are sufficiently open to be utilized.

Even if they were, you don't know that the specimen you hold in your hand, is not backdoored for sure without (possibly destructive) inspection.

Something like a logic gate package: it's so generic it'd be almost impossible to practically back-door it.
A CPLD or FPGA: could be more easily back-doored, but again, you'd maybe spot it in an X-ray photo since it'd be a "different" looking block inside a very regular and repetitive grid of blocks that have a very generic function and behaviour.
An ASIC like an off-the-shelf CPU (any architecture you like): all bets are off.

If you want fast: go x86, go modern ARM… maybe look at RISC-V → accept there may be backdoors.
If you want back-door free: get out your breadboards and start wiring TTL chips together → accept 8-bit or 16-bit CPUs with clock speeds in the kilohertz.

**Comment by rmsthebest** [comment]
Please stay on topic, I want to stay subscribed to this issue so I get notified when coreboot is available.

**Comment by hicaru** [comment]
> Okay. coreboot doesn't do anything different with the PSP than AMI does.

These are just words, and you've already misled me once. How am I supposed to believe in this now?  Т_Т

**Comment by Sean-StarLabs** [comment]
> How am I supposed to believe in this now?

Do you own research, PSP runs before the bootloader - AMD don't try and hide that

**Comment by hicaru** [comment]
> > How am I supposed to believe in this now?
> 
> Do you own research, PSP runs before the bootloader - AMD don't try and hide that

Friend, I understand all of this, and my point is really about something else. You could have mentioned that you support Coreboot so that people know what they're buying. My case is a prime example. I purchased something that didn't meet my expectations, and part of the mistake is mine—I could have checked the repository myself to see all the supported models! So, I'm heartbroken and left with no choice but to wait.

**Comment by hicaru** [comment]
Hello team. i see you still have there wrong info! please remove it! people will buy your laptop and get no coreboot! 
![image](https://github.com/StarLabsLtd/firmware/assets/21293723/b63864af-e79c-4b7b-9227-76d1b2edac3c)

**Comment by Sean-StarLabs** [comment]
"No firmware is available for this device." - you don't think that covers it?

**Comment by hicaru** [comment]
> "No firmware is available for this device." - you don't think that covers it?

ahh so i can download it and install directly SPI?

**Comment by Sean-StarLabs** [comment]
> ahh so can i download it and install directly SPI? could you share me a link for rom? or source code

I don't follow?

**Comment by hicaru** [comment]
> > ahh so can i download it and install directly SPI? could you share me a link for rom? or source code
> 
> I don't follow?

i mean if you have coreboot for this device, how can i build it? i want to flash SPI on board.

**Comment by Sean-StarLabs** [comment]
Click on the link - it shows "No firmware is available for this device."

**Comment by hicaru** [comment]
> Click on the link - it shows "No firmware is available for this device."

ok, i though i didn't get something, because there check the box opposite coreboot(LVFS).

**Comment by hicaru** [comment]
> Click on the link - it shows "No firmware is available for this device."

Please could you explain current stage, coreboot for VI AMD? sorry but we wait so many time  T_T

**Comment by Sean-StarLabs** [comment]
Working on it

**Comment by hicaru** [comment]
> on

you have an estimated release date?

**Comment by Sean-StarLabs** [comment]
I do not

**Comment by hicaru** [comment]
hello dear developers starlabs, i just have learned this page https://doc.coreboot.org/soc/amd/family17h.html, and i got a question! Why you cannot use open source `AGESA` which called `Picasso` in `soc/amd/picasso`? As i understand the Official AMD `AGESA` need NDA, you will wait NDA files endlessly, because they will never open them in near future.

**Comment by Sean-StarLabs** [comment]
That's a different processor, and would still have the same problem.

NDAs aren't in the equation either.

**Comment by hicaru** [comment]
> That's a different processor, and would still have the same problem.
> 
> NDAs aren't in the equation either.

yes, you are right((( it so sad

**Comment by cyb3rmonk** [comment]
I just realized I'm unable to update the system firmware to the latest version (i.e. `fwupdmgr refresh --force && fwupdmgr update`). I get this error in dmsg:

```
[   15.913653] systemd[1]: Starting Firmware update daemon...
[   15.994985] fwupd[1944]: 03:12:52.955 FuPluginFlashrom     FCH device found but SMBus revision 0x51 does not match known values.
[   15.994985] fwupd[1944]: Please report this to flashrom@flashrom.org and include this log and
[   15.994985] cyb3rlab2305 fwupd[1944]: the output of lspci -nnvx, thanks!.
[   15.994985] cyb3rlab2305 fwupd[1944]: 03:12:52.955 FuPluginFlashrom     Could not determine chipset generation.
[   15.995368] cyb3rlab2305 fwupd[1944]: 03:12:52.955 FuPluginFlashrom     Laptops, notebooks and netbooks are difficult to support and we
[   15.995368] fwupd[1944]: recommend to use the vendor flashing utility. The embedded controller
[   15.995368] fwupd[1944]: (EC) in these machines often interacts badly with flashing.
[   15.995368] fwupd[1944]: See the manpage and https://flashrom.org/Laptops for details.
[   15.995368] fwupd[1944]: If flash is shared with the EC, erase is guaranteed to brick your laptop
[   15.995368] fwupd[1944]: and write may brick your laptop.
[   15.995368] fwupd[1944]: Read and probe may irritate your EC and cause fan failure, backlight
[   15.995368] fwupd[1944]: failure and sudden poweroff.
[   15.995368] fwupd[1944]: You have been warned.
```

Is this caused by the the same PSP / Ap base issue? Should I open a new issue for this?

**Comment by Sean-StarLabs** [comment]
> Is this caused by the the same PSP / Ap base issue?

No, doesn't look like anything is wrong - what firmware are you running?

**Comment by cyb3rmonk** [comment]
```
# fwupdmgr get-devices
├─System Firmware:
│ │   Device ID:          c738f5c2f089f407525d70d40c11ec7dbeb1e687
│ │   Summary:            UEFI System Resource Table device (updated via NVRAM)
│ │   Current version:    1.2.0
│ │   Minimum Version:    1.2.0

# fwupdmgr refresh --force && fwupdmgr update
Updating lvfs
Downloading…
Successfully downloaded new metadata: 2 local devices supported
Devices with no available firmware updates: 
 • Star Drive PCIe SSD
╔════════════════════════════════════════════════════════════════╗
║ Upgrade System Firmware from 1.2.0 to 24.3.0?                  ║
╠════════════════════════════════════════════════════════════════╣
║ This release includes the following features:                  ║
║                                                                ║
║ • Fixed an issue that stopped the backlight working after S3   ║
║ • Enabled support for PD 2.0 chargers                          ║
║ • Disable USB power in S3                                      ║
║ • Improved battery reporting                                   ║
║                                                                ║
║ StarBook must remain plugged into a power source for the       ║
║ duration of the update to avoid damage.                        ║
╚════════════════════════════════════════════════════════════════╝
Perform operation? [Y|n]: y
Waiting…
Successfully installed firmware
Do not turn off your computer or remove the AC adapter while the update is in progress.
Devices with the latest available firmware version:
 • UEFI dbx
An update requires a reboot to complete. Restart now? [y|N]: y
```

Then I reboot and I get the logs above

**Comment by cyb3rmonk** [comment]
I don't want to spam everyone who's subscribed to this thread, so since you've said this wasn't related to the PSP / Ap base issue I opened a separate issue:

https://github.com/StarLabsLtd/firmware/issues/165

**Comment by Sean-StarLabs** [comment]
![20240611_113307](https://github.com/StarLabsLtd/firmware/assets/38053956/3f23381e-9355-428a-bbf9-64811df5eaa9)

Geting close...

**Comment by sysnux** [comment]
Great, thanks for your commitment to that issue!

**Comment by hicaru** [comment]
hello dear team, any news?

**Comment by Sean-StarLabs** [comment]
Nothing yet

**Comment by lil0r** [comment]
With the new starbook being released, does that mean we can finally wave good bye the hope to any descend support for the AMD machines?

**Comment by Sean-StarLabs** [comment]
No - they're completely unrelated..

**Comment by hicaru** [comment]
> No - they're completely unrelated..

When coreboot for AMD bro?

**Comment by lil0r** [comment]
yes, when coreboot bro? Nearing fast two year wait for your supporters.

**Comment by samueldr** [comment]
I know I'm not a "stakeholder" in this organization or repository. But let's act civil. Being rude is unlikely to help things get going, or worse: will annoy or anger the other party into not responding.

I know that, as an OSS developer, this kind of nagging might even make me de-prioritize the issue, if it's not one for me personally.

* * *

Here's my guess of how things are going. Anyone that knows: please correct me.

One is to assume they have a limited set of developers working on platform firmware [BIOS, UEFI, whatever you want to call it]. After all, only three contributors in this repo.

Couple that with the recent starlite launch (#184), development time was invested there in the past two months.

So, *even if there weren't any more blockers*, it's a good guesstimation that no time was available for this.

* * *

Here's how I'd approach it if I wanted to know about the state of things here:

> Hi!
>
> Since June/July 13th, has there been any change on things?
>
> While not asking for dates or time estimate, is there a list of things you can share that still needs to happen for Coreboot on the AMD variant? Of special interest is if there is anything blocked outside of StarLabs' control, to help better understand.
>
> Is there anything that can be done to help with this process?
>
> I am eager to make use of the Coreboot release to customize my personal system in a deeper manner.
>
> Cheers!

... or thereabout.

Which uh, I guess this is me asking, since I do have an AMD variant needing its Coreboot fulfilled :).

Cheers!

**Comment by hicaru** [comment]
> I know I'm not a "stakeholder" in this organization or repository. But let's act civil. Being rude is unlikely to help things get going, or worse: will annoy or anger the other party into not responding.
> 
> I know that, as an OSS developer, this kind of nagging might even make me de-prioritize the issue, if it's not one for me personally.
> 
> Here's my guess of how things are going. Anyone that knows: please correct me.
> 
> One is to assume they have a limited set of developers working on platform firmware [BIOS, UEFI, whatever you want to call it]. After all, only three contributors in this repo.
> 
> Couple that with the recent starlite launch (#184), development time was _likely_ invested there in the past two months.
> 
> So, _even if there weren't any more blockers_, it's a good guesstimation that no time was available for this.
> 
> Here's how I'd approach it if I wanted to know about the state of things here:
> 
> > Hi!
> > Since June/July 13th, has there been any change on things?
> > While not asking for dates or time estimate, is there a list of things you can share that still needs to happen for Coreboot on the AMD variant? Of special interest is if there is anything blocked outside of StarLabs' control, to help better understand.
> > Is there anything that can be done to help with this process?
> > I am eager to make use of the Coreboot release to customize my personal system in a deeper manner.
> > Cheers!
> 
> ... or thereabout.
> 
> Which uh, I guess this is me asking, since I do have an AMD variant needing its Coreboot fulfilled :).
> 
> Cheers!

it all sounds great, but in 2 years we only got a screenshot, I bought this laptop where there was a statement that it would work on coreboot, but it turned out not to be so! I feel cheated and if things are somehow difficult, I think the team could unsubscribe and maybe someone from the community will help in any way they can

**Comment by sjlongland** [comment]
If you can do it faster, knock yourself out… but bellyaching here is not going to make it happen any sooner.

**Comment by hamishfor** [comment]
Hi @Sean-StarLabs et all, is there any additional pressure you can apply to AMD? Sometimes the squeaky wheel... Either trying to get the official product manager on the phone, or another attempt at contacting the engineering team directly. It's really disappointing, I have a lot of AMD hardware.

**Comment by Sean-StarLabs** [comment]
> is there any additional pressure you can apply to AMD?

Short answer, no.

AFAIK, the OSS/coreboot stuff for AMD has been stopped because of some contract/products that got cancelled by Google. The engineers they hired to work on coreboot have been "repurposed".

**Comment by rmsthebest** [comment]
Does that mean coreboot is never happening?
Will you offer refunds for us who ordered coreboot laptops?

**Comment by Sean-StarLabs** [comment]
No, sorry - not sure how that read as it's off the table - I was just replying to @hamishfor who was asking if we could ask AMD for help

**Comment by Sean-StarLabs** [comment]
> Geting close...

Yes/no - the part that was blocking it is no longer blocking it, but we've got a lot that needs adjusting because of the way it was done

**Comment by arnytrty** [comment]
Hello, any updates?

**Comment by Sean-StarLabs** [comment]
Only one IRQ stopping Linux from booting - once that's done, need to get wireless, EC driver and a USB port working, but none of those three things should be too bad.

**Comment by kostadinnm** [comment]
The new development is nice. Here's what I bumped on regarding coreboot and a fairly similar to SLs vendor based in the US called MALIBAL: [Laptop Vendor MALIBAL Suggests Not Supporting Coreboot](https://www.phoronix.com/news/Malibal-Suggests-No-Coreboot). There must be something wrong in all that interaction. The fact that for such technical manners actors called "consultants" are involed makes me wonder of a very top executive fault in all this..

**Comment by Sean-StarLabs** [comment]
> a fairly similar to SLs vendor based in the US called MALIBA

Ouch

I know two of the three developers who were engaged for that work... there's enough information online to determine where the problem originates.

**Comment by sjlongland** [comment]
> The new development is nice. Here's what I bumped on regarding coreboot and a fairly similar to SLs vendor based in the US called MALIBAL: [Laptop Vendor MALIBAL Suggests Not Supporting Coreboot](https://www.phoronix.com/news/Malibal-Suggests-No-Coreboot). There must be something wrong in all that interaction. The fact that for such technical manners actors called "consultants" are involed makes me wonder of a very top executive fault in all this..

That is very disappointing news to be honest.  Sadly though, some people in the open-source community are like that. (e.g. there have been quite damning revelations about Richard Stallman of late… but I digress…)

My only other place I've used CoreBoot was on a PC Engines APU3… and there I loved it.  Especially on a headless box, it worked well.  Hence I'm eager to try it on a laptop.

If we're close to having something similar on the AMD-based StarBooks, then I wouldn't let this derail it to be honest.  There's still value in doing this, and I appreciate the work being done.

**Comment by kostadinnm** [comment]
I've just read the whole article from the Malibal's blog and I think those guys are overreacting by banning so many entities the cherry ot top being the whole state of Texas. I won't even link it here, because it's childish to the skies and won't help anyone.
Sean, could you point me where to look at, if I need to learn more about that fiasco? I guess something may pop up from phronix's article comments, which I am about to read tonight, but your input will be welcome.

**Comment by Sean-StarLabs** [comment]
Google search does it - mostly Reddit and forums where people post their interactions. I don't think you'll find anything public from the developers involved though.

**Comment by hamishfor** [comment]
I wouldn't read too much into anything Malibal says. They appear to butt heads with almost every vendor they work with. In all these childish fights, the only common denominator here, is Malibal.

Eagerly awaiting your future updates on Coreboot @Sean-StarLabs. Please proactively keep us as updated as you can! Thanks.

**Comment by hicaru** [comment]
any news?

**Comment by Sean-StarLabs** [comment]
Not yet

**Comment by hicaru** [comment]
> Not yet

2 years no news.... I bough the laptop with rial price 300$ for 1400$.... When I bough it there was written `coreboot` I got AMI bios and even cannot flash it by myself...

**Comment by rmsthebest** [comment]
> Linux boots - only partial graphics, no legacy IRQs, no WiFi.
> 
> > 2 years no news.... I bough the laptop with rial price 300$ for 1400$.... When I bough it there was written coreboot I got AMI bios and even cannot flash it by myself...
> 
> I don't follow this - you only got yours 10 months ago, and the price has never been $300, the processor to this day still costs more than $300, your order didn't have coreboot (the website was updated to make it super clear on the status) and you chose not to return it. Either way, as multiple users have asked above, please keep this on point - if you want to talk about anything with us, you can reach us at [support@starlabs.systems](mailto:support@starlabs.systems).

I'm a little sorry to further derail the comments, however...

I pre-ordered my laptop (the shipment details had coreboot checked) and once it arrived I contacted support since the keyboard had problems. When I sent it back I asked that you trade it in for a coreboot enabled intel one. This request was ignored, and the original laptop was returned with a repaired keyboard.

Anyway I wish you best of luck on the development efforts.

**Comment by hicaru** [comment]
> Linux boots - only partial graphics, no legacy IRQs, no WiFi.
> 
> > 2 years no news.... I bough the laptop with rial price 300$ for 1400$.... When I bough it there was written coreboot I got AMI bios and even cannot flash it by myself...
> 
> I don't follow this - you only got yours 10 months ago, and the price has never been $300, the processor to this day still costs more than $300, your order didn't have coreboot (the website was updated to make it super clear on the status) and you chose not to return it. Either way, as multiple users have asked above, please keep this on point - if you want to talk about anything with us, you can reach us at [support@starlabs.systems](mailto:support@starlabs.systems).

issue created Jan 29, 2023, regarding the return, you wrote that it would be at my expense! and When I bought it there was option with coreboot, as you answer on email that's laptop is supporting coreboot. just a scam for 1400$ possible buy Thinkpad and install there coreboot manually! I do not advise anyone to buy from this starlabs vendor!

**Comment by hamishfor** [comment]
> Linux boots - only partial graphics, no legacy IRQs, no WiFi.

Booting! Fantastic, thanks Sean, we all appreciate your work! Graphics might be prove to be problematic, but Wifi and legacy interrupts is a matter of working through each?

**Comment by Sean-StarLabs** [comment]
Pretty much

**Comment by arnytrty** [comment]
Hello, could you please provide an update on the progress? When can we expect the firmware to be ready? Thanks.

**Comment by lil0r** [comment]
Happy two year anniversary on this issue!
I guess I will have a new laptop before this will be resolved.

**Comment by Sean-StarLabs** [comment]
Little progress recently, but we just hired a new dev who used to work for AMD so things hopefully should speed up!

**Comment by hamishfor** [comment]
Hey @Sean-StarLabs - just wondering if we could get a little update on progress and hopefully the new starter has this as one of his priorities?

**Comment by mrusme** [comment]
Without disrespect to any one person specifically, but the amount of _"Are we there yet?"_ comments that this issue receives is somewhat annoying. I initially subscribed to this issue because I wanted to **receive updates from Star Labs** on this topic. While I understand everyone's frustration about a feature that years later still didn't materialize -- we're all in the same boat here! -- it should be clear that asking for an update every other week is unlikely to make it happen any faster. Star Labs are clearly struggling to get this done and it doesn't seem like users commenting here have the intention to offer their support.

@Sean-StarLabs if I might, I'd suggest ...

- ... either lock this issue (bottom right _"Lock conversation"_) so that only contributors to this repository can comment, to provide future updates
- ... or tell a different way to get informed whenever there are status updates on the Coreboot issue available (dedicated tech newsletter/mailing list?), for everyone who would wish to unsubscribe from this issue

Thank you.

**Comment by Sean-StarLabs** [comment]
@mrusme Good idea, thanks

@hamishfor AMI firmware was bought inline with coreboot as much as possible (side note, that'll go stable today/tomorrow) to make it easier to debug the difference as everything "looks" correct in coreboot

**Comment by Sean-StarLabs** [comment]
Fixed with 947fc5933dd62977d50d9e00cf596469fe658d4d

# 1. Check what you are running:

```bash
cat /sys/class/dmi/id/bios_vendor
cat /sys/class/dmi/id/bios_version
```

Expected values:
- AMI path: `American Megatrends International, LLC.` and `26.04`
- coreboot path: `coreboot` and `26.04`

# 2. If you are not already on AMI `26.04`, update to that first:

```bash
curl -fsSL https://github.com/StarLabsLtd/firmware/raw/refs/heads/capsules/updater.sh | bash
```

Then re-check:

```bash
cat /sys/class/dmi/id/bios_version
```

# 3. Switch to the coreboot beta:

```bash
curl -fsSL https://github.com/StarLabsLtd/firmware/raw/refs/heads/capsules/updater.sh | bash -s -- --coreboot-switch
```

This is only allowed on `B6-A`, and only from AMI `26.04`.

When it finishes, the updater will shut the machine down automatically.
After shutdown:
- disconnect the charger
- wait about 12 seconds until the LEDs flicker
- power it back on

# 4. Verify you are on coreboot:

```bash
cat /sys/class/dmi/id/bios_vendor
cat /sys/class/dmi/id/bios_version
```

Expected:
- `coreboot`
- `26.04`

# 5. Switch back to the AMI:

```bash
curl -fsSL https://github.com/StarLabsLtd/firmware/raw/refs/heads/capsules/updater.sh | bash -s -- --reinstall
```

That reflashes the normal `B6-A.bios` image and shuts the machine down automatically when complete.

# 6. Verify you are back on AMI:

```bash
cat /sys/class/dmi/id/bios_vendor
cat /sys/class/dmi/id/bios_version
```

Expected:
- `American Megatrends International, LLC.`
- `26.04`

**Comment by rmsthebest** [comment]
Thank you @Sean-StarLabs ! Wow what a journey it has been :)

I can confirm the switch works.

My only note to others is that at least for me the leds don't flash after shutdown, but the update are still done. I get a loud electrical pop sound instead. It's the same sounds I get if I press and quickly release the power button, the laptop then makes this sound and doesn't actually boot.

**Comment by kostadinnm** [comment]
Too bad my unit died(refused to boot) a couple of months ago(around Fall 2025). I extracted the ram, the ssd and the wifi adapter out. If anyone needs it, I am eager to send it to them free of charge(only transport expenses should be covered by the recipient). I made a couple of pictures before putting it back into the original packaging, so don't hesitate to ask for those beforehand.

**Comment by Sean-StarLabs** [comment]
> refused to boot

Sounds like linux bug, that was worked around in later versions of AMI. disconnected power, then powering on from the battery brought them back

**Comment by grisu48** [comment]
Confirmed working as well here, thank you this is awesome 🙂

**Comment by sjlongland** [comment]
I'll probably give this a try in the next day or so.  Is there a particular version of Linux and related tools we should do this from?  (I still have the factory Ubuntu install, release 24.04 still I think, just a matter of whether I do an update on it first or not.)

**Comment by Sean-StarLabs** [comment]
> Is there a particular version of Linux and related tools we should do this from?

Not really; only thing that'll likely struggle is Qubes

**Comment by sjlongland** [comment]
Turns out, it was Ubuntu 22.04 that I did the update from.  All worked.

![Starbook VI with Coreboot](https://github.com/user-attachments/assets/e301a8ce-7943-4aac-bd34-d2ee1408c612)

Many thanks for making this happen.

**Comment by mrusme** [comment]
I don't see AMI 26.04 via fwupdmgr, is this intentional?

**Comment by Sean-StarLabs** [comment]
Yes

**Comment by sjlongland** [comment]
Out of interest… now that I have Coreboot on here, I don't want it overwritten or tampered with by malware.  When IBM clones went to using flash chips for the BIOS, they had a write-project jumper that you had to remove before you could do a BIOS update, most of the time you left the jumper in place to prevent malware messing with it.

Is there a write-protect jumper on the Starbook I can set to protect it from unintentional manipulation?

**Comment by Sean-StarLabs** [comment]
> Is there a write-protect jumper on the Starbook I can set to protect it from unintentional manipulation?

No - things have moved on a little ;)

Rom Armor 2 is what's available - needs work to be usable in coreboot

**Comment by cyb3rmonk** [comment]
This no longer works
```
curl -fsSL https://github.com/StarLabsLtd/firmware/raw/refs/heads/capsules/updater.sh | bash -s -- --coreboot-switch
```

I notice that the coreboot switch was removed after moving to 26.06:
https://github.com/StarLabsLtd/firmware/commit/c81f2e1352021dae9bf8c5a10720557c80cc2577

What's the situation?

EDIT: I was able to do it by downloading the updater.sh from commit 7b8b17ca47cbcfcbc2c68b9e1da6a090f4f2c30b, then changing `REPO=` url to `firmware/7b8b17ca47cbcfcbc2c68b9e1da6a090f4f2c30b` instead of `firmware/capsules`. Maybe the fix here is just to provide a branch with this kind of workaround?

**Comment by mrusme** [comment]
@cyb3rmonk not sure but I could imagine that [this](https://github.com/StarLabsLtd/firmware/issues/396#issuecomment-4694922712) might be reason.

**Comment by Sean-StarLabs** [comment]
> @cyb3rmonk not sure but I could imagine that https://github.com/StarLabsLtd/firmware/issues/396#issuecomment-4694922712 might be reason.

It's not

> What's the situation?

`--coreboot-switch`just isn't there any more, so it's coreboot only going forward
